smart-metastore/src/main/java/org/smartdata/metastore/dao/accesscount/AccessCountTableManager.java

```diff
@@ -151,44 +154,220 @@ private void recoverTables() {
-          searchIntervalGranularity)) {
-
-        // skip interval if it ends before specified search start time
-        if (table.getEndTime() <= startTime) {
+      long finalStartTime = startTime;
+      Collection<AccessCountTable> tables = inMemoryTableManager.getTablesOfGranularity(
+              searchIntervalGranularity).stream()
+          .filter(t -> t.getStartTime() <= endTime)
+          .collect(Collectors.toList());
+      boolean foundTables = !tables.isEmpty();
+      if (parentTable == null && !decreaseOnly) {
+        //when we didn't find any tables of current granularity,
+        // we have to try to find tables of bigger granularity
+        while (!foundTables) {
+          searchIntervalGranularity = increaseGranularity(searchIntervalGranularity);
+          if (searchIntervalGranularity == null) {
+            tables = Collections.emptyList();
+            searchIntervalGranularity = startGranularity;
+            foundTables = true;
+          } else {
+            tables = inMemoryTableManager.getTablesOfGranularity(
+                searchIntervalGranularity);
+            Long accessCountTableStartTime = tables.stream()
+                .findFirst()
+                .map(AccessCountTable::getStartTime)
+                .orElse(Long.MAX_VALUE);
+            foundTables = accessCountTableStartTime < startTime;
+          }
+        }
+        decreaseOnly = true;
+      }
+      tables = tables.stream()
+          .filter(t -> {
+            if (parentTable != null) {
+              //include child tables which has endTime after startTime
+              // and includes into parent table
+              return t.getEndTime() > finalStartTime && t.getEndTime() <= parentTable.getEndTime();
+            }
+            return t.getEndTime() > finalStartTime;
+          })
+          .collect(Collectors.toList());
+      int n = 0;
+      for (AccessCountTable table : tables) {
+        n++;
+        //skip tables if it starts after specified startTime
+        if (table.getStartTime() > endTime) {
           continue;
         }
-
-        // the access count table interval is entirely in the search interval
-        if (table.getStartTime() >= startTime) {
-          results.add(table);
-          startTime = table.getEndTime();
+        if (table.getStartTime() <= startTime) {
+          if (table.getStartTime() == startTime) {
+            if (table.getEndTime() <= endTime) {
+              result.add(table);
+              startTime = table.getEndTime();
+            } else {
+              //table.endTime > endTime
+              if (searchIntervalGranularity != TimeGranularity.SECOND) {
+                List<AccessCountTable> childTables =
+                    getAccessCountTables(startTime, endTime, table);
+                result.addAll(childTables);
+              } else {
+                addPartialTable(table, table.getStartTime(), endTime)
+                    .ifPresent(result::add);
+              }
```

Comment:
This part of code is common for all cases except `table.getStartTime() == startTime && table.getEndTime() <= endTime`. Please, consider extracting it into separate function

Reply:
done